src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java

```diff
@@ -103,6 +110,10 @@ private void setFilingApiData(FilingApi filing, String requestId, String oversea
         boolean isUpdateFiling = setSubmissionData(data, overseasEntityId, logMap);
```

Comment:
Make setSubmissionData return the overseasEntitySubmissionDto. Then can use it to ask overseasEntitySubmissionDto.isForUpdate() and overseasEntitySubmissionDto.getEntityNumber().
Taking things out of hash maps is quite fragile.
Have a look at: https://github.com/companieshouse/overseas-entities-api/pull/210